scipy/signal/_filter_design.py

```diff
@@ -1214,13 +1214,21 @@ def tf2zpk(b, a):
         array([ -2.5+2.59807621j ,  -2.5-2.59807621j]),
         3.0)
     """
+    xp = array_namespace(b, a)
     b, a = normalize(b, a)
-    b = (b + 0.0) / a[0]
-    a = (a + 0.0) / a[0]
+
+    if xp.isdtype(b.dtype, 'integral'):
+        b = xp.astype(b, xp.float64)
+    if xp.isdtype(a.dtype, 'integral'):
+        a = xp.astype(a, xp.float64)
+
+    b = b / a[0]
```

Comment:
I've probably just forgotten but how come we can no longer do this in-place?

Reply:
a three-letter word: jax